toml/_parser.ts

```diff
@@ -121,6 +121,15 @@ export class Scanner {
   startsWith(searchString: string) {
     return this.#source.startsWith(searchString, this.#position);
   }
+
+  match(regExp: RegExp) {
+    regExp = new RegExp(
+      regExp.source,
+      regExp.sticky ? regExp.flags : regExp.flags + "y",
+    );
+    regExp.lastIndex = this.#position;
+    return this.#source.match(regExp);
```

Comment:
This matches against (potentially large) `#source` text. On the other hand, the current code usually slices only necessary small text before matching/testing against regexp. I'm not sure this doesn't include any perf degradation.